### ISSUE TITLE
VB nameof called with an byref param inside of a lambda fails.

### ISSUE THREAD

**Issue by TyOverby** [dotnet/roslyn#10839]
This is a sister-issue to #10467 which had the C# side fixed by https://github.com/dotnet/roslyn/pull/10776

Tracked separately because the VB fix is different.
### Repro as a test case

``` vb
        Public Sub NameOfLambda()
            Dim compilationDef =
                <compilation>
                    <file name="a.vb">
Module Program
    Sub DoSomething(ByRef x As Integer)
        Dim f = Function()
                    Return NameOf(x)
                End Function
        System.Console.WriteLine(f())
    End Sub
    Sub Main()
        Dim x =  5
        DoSomething(x)
    End Sub
End Module
                </file>
                </compilation>
            Dim comp = CreateCompilationWithMscorlibAndVBRuntime(compilationDef, TestOptions.DebugExe)
            CompileAndVerify(comp, expectedOutput:="x").VerifyDiagnostics()
        End Sub
```

**Comment by TyOverby** [comment]
Assigning to @AnthonyDGreen because he said that he wanted this bug for external developers.

**Comment by jcouv** [comment]
@AnthonyDGreen What's the plan for this bug? If this needs to be fixed in RTM, we need to assign to a dev asap.
@TyOverby Just to confirm, this currently reports an error (but no crash), whereas expected is that the compilation succeeds, right?

**Comment by TyOverby** [comment]
@jcouv: correct.  Does not compile, but it does error.

**Comment by jcouv** [comment]
Ping for @AnthonyDGreen 
What's the plan for this bug? If this needs to be fixed in RTM, we need to assign to a dev asap.

**Comment by TyOverby** [comment]
@jcouv I'll take this one.

**Comment by jcouv** [comment]
Tracked by VSO https://devdiv.visualstudio.com/DevDiv/_workitems/edit/369808

**Comment by TyOverby** [comment]
Closed by #16596
